### PR TITLE
Bump go version to 1.22.3 and update dependencies

### PR DESCRIPTION
**Pull request by rzetelskik**
**Description of your changes:** This PR bumps go version to 1.22.3 and updates dependencies to latest patch releases to fix [GO-2024-2824: A malformed DNS message in response to a query can cause the Lookup functions to get stuck in an infinite loop](https://pkg.go.dev/vuln/GO-2024-2824).

~It also updates the minor version of github.com/kubernetes-csi/csi-test/v5 to 5.2.0 - sanity tests were failing otherwise.~
Edit: reverted, they are flaky anyway.

**Which issue is resolved by this Pull Request:**
Resolves https://github.com/scylladb/k8s-local-volume-provisioner/issues/47

/kind feature
/priority critical-urgent
/cc zimnx

### CODE REVIEW

**Review by zimnx — DISMISSED** [review]
lgtm
/assign tnozicka

**Comment by rzetelskik** [comment]
@zimnx any idea why test-sanity could become flaky from just the dependency bump?

**Comment by rzetelskik** [comment]
> @zimnx any idea why test-sanity could become flaky from just the dependency bump?

I ran it a 100 times with github.com/kubernetes-csi/csi-test/v5 v5.2.0 and it passed every single time, but it failed on a 6th attempt with v5.1.0.

```
All tests passed...
This was attempt 5 of 101.
Running Suite: Sanity Suite - /home/rzetelskik/github.com/scylladb/k8s-local-volume-provisioner/test/sanity/set/localdriver
===========================================================================================================================
Random Seed: 1715185370

Will run 83 of 84 specs
••SSS
------------------------------
P [PENDING]
Local CSI Driver CSI sanity Controller Service [Controller Server] ListVolumes pagination should detect volumes added between pages and accept tokens when the last volume from a page is deleted
/home/rzetelskik/github.com/scylladb/k8s-local-volume-provisioner/vendor/github.com/kubernetes-csi/csi-test/v5/pkg/sanity/controller.go:268
------------------------------
•••••••SSSS•••••••SSSSSSSSSSSSSSSSSSSSSSSSSSSSS
------------------------------
• [FAILED] [60.053 seconds]
Local CSI Driver CSI sanity GroupController Service [GroupController VolumeGroupSnapshots] [BeforeEach] CreateVolumeGroupSnapshot should fail when no name is provided
  [BeforeEach] /home/rzetelskik/github.com/scylladb/k8s-local-volume-provisioner/vendor/github.com/kubernetes-csi/csi-test/v5/pkg/sanity/tests.go:46
  [It] /home/rzetelskik/github.com/scylladb/k8s-local-volume-provisioner/vendor/github.com/kubernetes-csi/csi-test/v5/pkg/sanity/groupcontroller.go:158

  Timeline >>
  STEP: connecting to CSI driver @ 05/08/24 18:22:51.088
  [FAILED] in [BeforeEach] - /home/rzetelskik/github.com/scylladb/k8s-local-volume-provisioner/vendor/github.com/kubernetes-csi/csi-test/v5/pkg/sanity/sanity.go:265 @ 05/08/24 18:23:51.131
  [PANICKED] in [AfterEach] - /usr/local/go/src/runtime/panic.go:261 @ 05/08/24 18:23:51.132
  << Timeline

  [FAILED] Unexpected error:
      <*errors.errorString | 0xc000689620>:
      Connection timed out
      {
          s: "Connection timed out",
      }
  occurred
  In [BeforeEach] at: /home/rzetelskik/github.com/scylladb/k8s-local-volume-provisioner/vendor/github.com/kubernetes-csi/csi-test/v5/pkg/sanity/sanity.go:265 @ 05/08/24 18:23:51.131

  There were additional failures detected.  To view them in detail run ginkgo -vv
------------------------------
SSSS•••••••••••SSSSSSSSSSSSS••

Summarizing 1 Failure:
  [FAIL] Local CSI Driver CSI sanity GroupController Service [GroupController VolumeGroupSnapshots] [BeforeEach] CreateVolumeGroupSnapshot should fail when no name is provided
  /home/rzetelskik/github.com/scylladb/k8s-local-volume-provisioner/vendor/github.com/kubernetes-csi/csi-test/v5/pkg/sanity/sanity.go:265

Ran 30 of 84 Specs in 60.860 seconds
FAIL! -- 29 Passed | 1 Failed | 1 Pending | 53 Skipped
--- FAIL: TestSanity (60.87s)
FAIL

Tests failed on attempt #6
```

Does this have a history of being flaky?
Even with the version bump, I'm skeptical of merging this anyway now.

File: go.mod [hunk 1]
```diff
@@ -1,38 +1,38 @@
 module github.com/scylladb/k8s-local-volume-provisioner
 
-go 1.21
+go 1.21.10
```
Comment (tnozicka) [inline go.mod:3]: Shouldn't we bump this to 1.22.3? I don't think we want to run on 2 separate go version modulo a transition period.

Comment (rzetelskik) [reply go.mod:3]: I can bump it to 1.22.3. I'll have to update CI first though. Sanity tests seem to be flaky on current master - I'll open a separate issue for it.

**Review by tnozicka — APPROVED** [review]
/approve
/lgtm
(with a followup for golang major bump)

**Comment by rzetelskik** [comment]
@tnozicka should this bump dependencies to newest ones as well (including k8s v1.30.0) or just bump patch releases?

**Comment by rzetelskik** [comment]
Sorry for accidental closing, I messed up rebasing in the process.

Sanity tests flake issue xref: https://github.com/scylladb/k8s-local-volume-provisioner/issues/49

**Comment by tnozicka** [comment]
patch release for kube is fine (minor is more pain to bump)

**Review by tnozicka — APPROVED** [review]
/lgtm

**Comment by rzetelskik** [comment]
> patch release for kube is fine (minor is more pain to bump)

I actually tried it and didn't run into any difficult issues other than the usual replacement bumps, but I'm happy to keep it as is to have it aligned with other repositories.

**Pull request merged**